nearcache/config.go

```diff
@@ -0,0 +1,256 @@
+	PreloaderConfig PreloaderConfig
+	// InMemoryFormat specifies how the entry values are stored in the Near Cache.
+	// InMemoryFormatBinary stores the values after serializing them.
+	// InMemoryFormatObject stores the values in their original form.
+	// The default is InMemoryFormatBinary.
+	InMemoryFormat InMemoryFormat
+	// SerializeKeys specifies how the entry keys are stored in the Near Cache.
+	// If false, keys are stored in their original form.
+	// If true, keys are stored after serializing them.
+	// Storing keys in serialized form is required when the key is not hashable, such as slices.
+	// The default is false.
+	SerializeKeys bool
+	// TimeToLiveSeconds is the maximum number of seconds for each entry to stay in the Near Cache (time to live).
+	// Entries that are older than TimeToLiveSeconds will automatically be evicted from the Near Cache.
+	// Must be non-negative.
+	// The value 0 equals to math.MaxInt32
+	// The default is 0.
+	TimeToLiveSeconds int
+	//MaxIdleSeconds is the maximum number of seconds each entry can stay in the Near Cache as untouched (not-read).
+	//Entries that are not read (touched) more than MaxIdleSeconds value will get removed from the Near Cache.
+	//Accepts any integer between {@code 0} and {@link Integer#MAX_VALUE}.
+	// Must be non-negative.
+	// The value 0 equals to math.MaxInt32
+	// The default is 0.
+	MaxIdleSeconds int
+}
+
+// Clone returns a copy of the configuration.
+func (c Config) Clone() Config {
+	return Config{
+		invalidateOnChange: c.invalidateOnChange,
+		Name:               c.Name,
+		EvictionConfig:     c.EvictionConfig.Clone(),
+		PreloaderConfig:    c.PreloaderConfig.Clone(),
+		InMemoryFormat:     c.InMemoryFormat,
+		SerializeKeys:      c.SerializeKeys,
+		TimeToLiveSeconds:  c.TimeToLiveSeconds,
+		MaxIdleSeconds:     c.MaxIdleSeconds,
+	}
+}
+
+// Validate validates the configuration and replaces missing configuration with defaults.
+func (c *Config) Validate() error {
+	if c.Name == "" {
+		c.Name = "default"
+	}
+	if c.TimeToLiveSeconds == 0 {
+		c.TimeToLiveSeconds = math.MaxInt32
+	}
+	if c.MaxIdleSeconds == 0 {
+		c.MaxIdleSeconds = math.MaxInt32
+	}
+	if err := c.EvictionConfig.Validate(); err != nil {
+		return err
+	}
+	if err := c.PreloaderConfig.Validate(); err != nil {
+		return err
+	}
+	if _, err := check.NonNegativeInt32(c.TimeToLiveSeconds); err != nil {
+		return ihzerrors.NewInvalidConfigurationError("nearcache.Config.TimeToLiveSeconds: out of range", err)
+	}
+	if _, err := check.NonNegativeInt32(c.MaxIdleSeconds); err != nil {
+		return ihzerrors.NewInvalidConfigurationError("nearcache.Config.MaxIdleSeconds: out of range", err)
+	}
+	return nil
+}
+
+/*
+SetInvalidateOnChange sets if Near Cache entries are invalidated when the entries in the backing data structure are changed (updated or removed).
+When this setting is enabled, a Hazelcast instance with a Near Cache listens for cluster-wide changes on the entries of the backing data structure.
```

Comment:
We can use "client" instead of Hazelcast instance as well.

```suggestion
When this setting is enabled, a client with a Near Cache listens for cluster-wide changes on the entries of the backing data structure.
```

Similar for "Changes done on the local Hazelcast instance" below

Reply:
Updated at: https://github.com/hazelcast/hazelcast-go-client/pull/786/commits/da609e576a21dc14356bf90a242e034cc8be605e

Reply:
Can you update `Changes done on the local Hazelcast instance always invalidate` sentence as well?

Reply:
Updated at: https://github.com/hazelcast/hazelcast-go-client/pull/786/commits/067807ba8f2dcd7e92d074d022493ed39c9efaaf